src/awst_build/eb/bytes-expression-builder.ts

```diff
@@ -325,3 +336,30 @@ export class ToStringBuilder extends ParameterlessFunctionBuilder {
     )
   }
 }
+
+class EqualsFunctionBuilder extends FunctionBuilder {
+  constructor(private expr: awst.Expression) {
+    super(expr.sourceLocation)
+  }
+
+  call(args: ReadonlyArray<NodeBuilder>, typeArgs: ReadonlyArray<PType>, sourceLocation: SourceLocation): NodeBuilder {
+    const {
+      args: [right],
+    } = parseFunctionArgs({
+      args,
+      typeArgs,
+      genericTypeArgs: 0,
+      callLocation: sourceLocation,
+      funcName: 'equals',
+      argSpec: (a) => [a.required(bytesPType)],
+    })
+    return new BooleanExpressionBuilder(
+      nodeFactory.bytesComparisonExpression({
+        operator: EqualityComparison.eq,
+        lhs: this.expr,
+        rhs: right.toBytes(sourceLocation),
```

Comment:
`.resolve()` should suffice as `right` is already going to be of type bytes

Reply:
It has been updated in this commit, 6c31fad.